R/vulnerability/disasters-emergencies/flooding/2022-interim/england/direct-flood-experience/historic-flooding.R

```diff
@@ -0,0 +1,46 @@
+library(dplyr)
+library(sf)
+library(geographr)
+
+source("R/utils.R")  # for download_file()
+
+# Download historic flood map from https://environment.data.gov.uk/DefraDataDownload/?mapService=EA/HistoricFloodMap&Mode=spatial
+# Note: You may need to generate a new download URL (below) from the link above
+tf <- download_file("https://environment.data.gov.uk/UserDownloads/interactive/015ae0b56fe64bfd8ed0a6108107e56670824/EA_HistoricFloodMap_GeoJSON_Full.zip", ".zip")
+
+unzip(tf, exdir = tempdir())
+
+historic_flooding <- read_sf(file.path(tempdir(), "data", "Historic_Flood_Map.json"))
+
+# ---- Load Output Area boundaries ----
+# Source: https://geoportal.statistics.gov.uk/datasets/ons::output-areas-december-2011-boundaries-ew-bfe/about
+# Opening straight from the web takes far too long, so I manually downloaded a local copy and will load from that
+# boundaries_oa <- read_sf("https://services1.arcgis.com/ESMARspQHYMw9BZ9/arcgis/rest/services/Output_Areas_December_2011_Boundaries_EW_BFE/FeatureServer/0/query?where=1%3D1&outFields=OA11CD&outSR=4326&f=json")
+boundaries_oa <- read_sf("data/vulnerability/disasters-emergencies/flooding/Output_Areas__December_2011__Boundaries_EW_BFE.shp")
+
+# Look up which Output Areas are in historical flooding areas
+flood_oa_shp <- 
+  boundaries_oa %>% 
+  st_join(historic_flooding)
```

Comment:
I think it might automatically do this when joining 2 datasets with different crs but in similar analysis I did from D&E explicitly converted the OA boundaries data from the lat/long system to a 2D system used for joining (which the flooding data is already in the 2D projected system) https://github.com/britishredcrosssociety/resilience-index/blob/3bacd23645483be2c8e3950032257cc871cdb318/R/shocks/disasters-emergencies/england/flood.R#L42 so could be worth potentially adding in:

```suggestion
flood_oa_shp <- 
  boundaries_oa %>% 
  st_transform(crs = st_crs(historic_flooding)) %>%
  st_join(historic_flooding)
  
```

Reply:
Nice idea. The CRSs of those two datasets are the same anyway (at least when I run the code...) but definitely a good idea to make sure.